src/resources/js/components/global/taskComponents/AddTaskData.vue

```diff
@@ -356,19 +377,32 @@ export default {
             });
         }, 500),
 
-        updateTaskData(event) {
+        updateTaskDataInTaskModal(event) {
             let isValid = this.validateCreateOrUpdateTaskEvent(this.cloneCardData, this.checkedOptions)
 
             if (isValid) {
                 event.target.disabled = true;
-                this.eventHub.$emit("update-task-card-data", this.cloneCardData);
-                this.eventHub.$emit("close-task-modal");
+
+                if (this.source === "KanbanTaskModal") {
+                    this.eventHub.$emit("update-task-card-data", this.cloneCardData);
+                    this.eventHub.$emit("close-task-modal");
+                }
+                if (this.source === "TaskPane") {
+                    this.eventHub.$emit("update-task-card-data", this.cloneCardData);
+                }
             }
         },
 
-        cancel(event) {
-            event.target.disabled = true;
-            this.eventHub.$emit("close-task-modal");
+        cancel() {
+
+            if (this.source === "KanbanTaskModal") {
```

Comment:
Consider an enum file for your kanban component names

Reply:
fuck, yeah i'll do that